extensions/sql-database-projects/src/controllers/projectController.ts

```diff
@@ -649,134 +607,64 @@ export class ProjectsController {
 		return treeNode instanceof FolderNode ? utils.trimUri(treeNode.root.uri, treeNode.uri) : '';
 	}
 
-	public async getModelFromContext(context: any): Promise<ImportDataModel | undefined> {
-		let model = <ImportDataModel>{};
+	/**
+	 * Creates a new SQL database project from the existing database,
+	 * prompting the user for a name, file path location and extract target
+	 */
+	public async createProjectFromDatabase(context: azdata.IConnectionProfile | any): Promise<CreateProjectFromDatabaseDialog> {
+		const profile = this.getConnectionProfileFromContext(context);
+		let createProjectFromDatabaseDialog = this.getCreateProjectFromDatabaseDialog(profile);
 
-		let profile = this.getConnectionProfileFromContext(context);
-		let connectionId, database;
-		//TODO: Prompt for new connection addition and get database information if context information isn't provided.
+		createProjectFromDatabaseDialog.createProjectFromDatabaseCallback = async (model) => await this.createProjectFromDatabaseCallback(model);
 
-		if (profile) {
-			database = profile.databaseName;
-			connectionId = profile.id;
-		}
-		else {
```

Comment:
if profile is undefined, are we not prompting user to open a connection anymore?

Reply:
No. It will be done in the UI itself, where the server can be connected, in case profile is undefined. If the creation happens from within database (or server), that information is pre-populated.